src/core/catalog.js

```diff
@@ -486,17 +486,17 @@ class Catalog {
         return shadow(this, "optionalContentConfig", null);
       }
       const groups = [];
-      const groupRefs = new RefSet();
+      const groupsMap = new Map();
       // Ensure all the optional content groups are valid.
       for (const groupRef of groupsData) {
-        if (!(groupRef instanceof Ref) || groupRefs.has(groupRef)) {
+        if (!(groupRef instanceof Ref) || groupsMap.has(groupRef)) {
           continue;
         }
-        groupRefs.put(groupRef);
-
-        groups.push(this.#readOptionalContentGroup(groupRef));
+        const group = this.#readOptionalContentGroup(groupRef);
+        groups.push(group);
+        groupsMap.set(groupRef, group);
       }
-      config = this.#readOptionalContentConfig(defaultConfig, groupRefs);
+      config = this.#readOptionalContentConfig(defaultConfig, groupsMap);
```

Comment:
The reason why I am using `Map` instead of `RefSetCache` is that I need its `keys()` method which the latter does not provide:  `groupsMap` is passed to `#readOptionalContentConfig` on line 499. Then, in the local function `parseOrder`, I want to iterate over the keys when the `hiddenGroups` array is filled.

Reply:
https://github.com/mozilla/pdf.js/blob/e25832c43fe4d945b0773aef9abf7d8a6325c34f/src/core/primitives.js#L386-L390

Reply:
I need some more help on this as I still don't know how to translate the `Map`-based to a `SetRefCache-based` formulation (function parseOrder) . None of 

````
const groupRef of contentGroupsMap.keys() -->  const groupRef of groupRefCache.items().keys()
const groupRef of contentGroupsMap.keys() -->  const groupRef of groupRefCache.items()
````
works.

Reply:
```javascript
for (const [groupRef] of contentGroupRefCache.items()) {
  ...
}
```

Reply:
Thanks! I did it slightly differently in the meantime. I updated the code once more using your solution.